Default `clang` version is outdated

The latest stable release of macOS (`13.3.1`) ships `clang` version `14.0.3`, but our current default is still `14.0.0`, thus breaking the `bazel build :urbit` command on `macos-aarch64` computers that don't pass the extra `--clang_version=14.0.3` flag.

I get a failure building on the latest stable release of macOS even with the clang_version flag set:

```
2 warnings generated.
INFO: From Compiling deps/libyrmcds/counter.c:
external/h2o/deps/libyrmcds/counter.c:212:18: warning: implicit conversion loses integer precision: 'yrmcds_cnt_command' to 'uint8_t' (aka 'unsigned char') [-Wimplicit-int-conversion]
    r->command = (yrmcds_cnt_command)c->recvbuf[1];
               ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/h2o/deps/libyrmcds/counter.c:213:17: warning: implicit conversion loses integer precision: 'yrmcds_cnt_status' to 'uint8_t' (aka 'unsigned char') [-Wimplicit-int-conversion]
    r->status = (yrmcds_cnt_status)c->recvbuf[2];
              ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
ERROR: /private/var/tmp/_bazel_loganallen/f70191a3f2d056a5650de01a33518124/external/zlib/BUILD.bazel:3:11: Compiling gzlib.c failed: (Exit 1): clang failed: error executing command (from target @zlib//:zlib) /usr/bin/clang -MD -MF bazel-out/darwin_arm64-fastbuild/bin/external/zlib/_objs/zlib/gzlib.d '-frandom-seed=bazel-out/darwin_arm64-fastbuild/bin/external/zlib/_objs/zlib/gzlib.o' ... (remaining 16 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
external/zlib/gzlib.c:252:9: error: call to undeclared function 'lseek'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        LSEEK(state->fd, 0, SEEK_END);  /* so gzoffset() is correct */
        ^
external/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
external/zlib/gzlib.c:252:9: note: did you mean 'fseek'?
external/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:162:6: note: 'fseek' declared here
int      fseek(FILE *, long, int);
         ^
external/zlib/gzlib.c:258:24: error: call to undeclared function 'lseek'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        state->start = LSEEK(state->fd, 0, SEEK_CUR);
                       ^
external/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
external/zlib/gzlib.c:359:9: error: call to undeclared function 'lseek'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    if (LSEEK(state->fd, state->start, SEEK_SET) == -1)
        ^
external/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
external/zlib/gzlib.c:400:15: error: call to undeclared function 'lseek'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        ret = LSEEK(state->fd, offset - state->x.have, SEEK_CUR);
              ^
external/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
external/zlib/gzlib.c:496:14: error: call to undeclared function 'lseek'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    offset = LSEEK(state->fd, 0, SEEK_CUR);
             ^
external/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
5 errors generated.
Target //pkg/vere:urbit failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 4.205s, Critical Path: 3.36s
INFO: 488 processes: 156 internal, 332 darwin-sandbox.
```

--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -44,8 +44,8 @@ config_setting(
 # Version flag for clang.
 string_flag(
     name = "clang_version",
-    # macOS uses `clang-14.0.0` by default.
-    build_setting_default = "14.0.0",
+    # macOS uses `clang-14.0.3` by default.
+    build_setting_default = "14.0.3",
     visibility = ["//visibility:public"],
 )
 

